emulator/src/kreator/preprocessor.rs

```diff
@@ -393,6 +395,42 @@ fn get_labels(code: &Vec<String>) -> Result<HashMap<String, u16>, &'static str>
     Ok(labels)
 }
 
+fn get_opc_by_byte_size() -> (Vec<&'static str>, Vec<&'static str>, Vec<&'static str>) {
+    (vec![
+        "NOP", "STAX", "INX", "INR", "DCR", "RLC", "DAD", "LDAX", "DCX", "RRC", "RAL", "RAR",
+        "DAA", "STC", "CMC", "HLT", "SBB", "MOV", "ANA", "XRA", "ORA", "CMP", "RNZ", "POP",
+        "PUSH", "RST", "RZ", "RET", "RNC", "RC", "RPO", "XTHL", "RPE", "PCHL", "XCHG", "RP",
+        "DI", "RM", "SPHL", "EI", "CMA", "ADD", "ADC", "SUB"
+    ],
+    vec![
+        "MVI", "ADI", "ACI", "OUT", "SUI", "IN", "SBI", "ANI", "XRI", "ORI", "CPI"
+    ],
+    vec![
+        "LXI", "SHLD", "LHLD", "STA", "LDA", "JNZ", "JMP", "CNZ", "JZ", "CZ", "CALL", "JNC",
+        "CNC", "JC", "CC", "JPO", "CPO", "JPE", "CPE", "JP", "CP", "JM", "CM"
+    ])
+}
+
+fn get_byte_amount_of_line(line: &String) -> i32 {
+    let (one_byte_labels, two_byte_labels, three_byte_labels) = get_opc_by_byte_size();
+    for opc in &one_byte_labels {
+        if line.starts_with(opc) {
+            return 1;
+        }
+    }
+    for opc in &two_byte_labels {
+        if line.starts_with(opc) {
+            return 2;
+        }
+    }
+    for opc in &three_byte_labels {
+        if line.starts_with(opc) {
+            return 3;
+        }
+    }
+    0
```

Comment:
If this can't happen, maybe panic! or unreachable! it

Reply:
This can happen (and will happen) quite often. Because the labels have to be determined before the actual preprocessing lines like `ORG 10` or `IF 0` will still be in the code but should not affect the memory adress if my understanding is correct. So every line that is _not_ an instruction translated to byte code should simply have the value `0`